File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = [0]*len(prices)
    
    stack =[]
    
    for i in range(len(prices)):
        for idx, s in stack:
            answer[idx]+=1
        while stack and stack[-1][1] > prices[i]:
            stack.pop()
        stack.append((i, prices[i]))
    
    
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        int n= prices.length;
        int[] answer = new int[n];

        Stack<int[]> stack = new Stack<>();
        for(int i=0;i<n;i++){
            for(int[] s:stack){
                answer[s[0]] += 1;
            }
            while(!stack.isEmpty() && stack.peek()[1] > prices[i]){
                stack.pop();
            }
            stack.push(new int[]{i, prices[i]});
        }
        
        return answer;
    }
}